ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageCompactor.java

```diff
@@ -96,28 +99,65 @@ public int compactPartitions() {
     return totalBlobsPurged;
   }
 
+  /**
+   * Returns the expired blob in the specified partition with the earliest expiration time.
+   * @param partitionPath the partition to check.
+   * @return the {@link CloudBlobMetadata} for the expired blob, or NULL if none was found.
+   * @throws CloudStorageException
+   */
+  public CloudBlobMetadata getOldestExpiredBlob(String partitionPath) throws CloudStorageException {
+    List<CloudBlobMetadata> deadBlobs = requestAgent.doWithRetries(
+        () -> cloudDestination.getExpiredBlobs(partitionPath, System.currentTimeMillis(), queryLimit), "GetDeadBlobs",
```

Comment:
should the query limit be 1 here and in the method below since you are only interested in a single blob?

Reply:
Yes, changed.